test/integration/install/test_install.py

```diff
@@ -614,3 +615,43 @@ def test_install_tt_already_exists_with_symlink(tt_cmd, tmpdir):
     expected_bin = os.path.join(bin_path, "tt_v1.1.2")
     tarantool_bin = os.path.realpath(os.path.join(bin_path, "tt"))
     assert tarantool_bin == expected_bin
+
+
+@pytest.mark.parametrize("package_to_exclude, expected_rc", [
+    pytest.param(
+        "mage",
+        1,
+    ),
+    pytest.param(
+        "",
+        0,
+    )
+])
+def test_install_tt_exit_code_dependency_check(
+        tt_cmd,
+        tmpdir,
+        package_to_exclude,
+        expected_rc):
+    # Create new PATH with excluded packages
+    original_path = os.environ['PATH']
+    new_path_elements = []
+    for dir in original_path.split(os.pathsep):
+        if os.path.isdir(dir) and package_to_exclude not in os.listdir(dir):
+            new_path_elements.append(dir)
+    new_path = os.pathsep.join(new_path_elements)
+
+    with mock.patch.dict(os.environ, {'PATH': new_path}):
```

Comment:
No need patch existing env and restore it after `with`, just pass the new env to the `subprocess.Popen` call.

Reply:
Changed it.